src/Mod/Fem/Gui/ViewProviderFemPostObject.cpp

```diff
@@ -758,11 +759,35 @@ void ViewProviderFemPostObject::onSelectionChanged(const Gui::SelectionChanges &
     // color bar.
     // But don't do this if the object is invisible because other objects with a
     // color bar might be visible and the color bar is then wrong.
+    // For pipelines, hide other visible ones and make the current visible since there
+    // is technically no way to paint them over each other and showing the one one likes
+    // Corresponding info from Realthunder:
+    // "Because on top rendering delays rendering to the last, where the OpenGL depth buffer
+    // is already populated. On top rendering is done with depth write disabled and only with
+    // transparency painting on top is somewhat visually acceptable."
     if (sel.Type == sel.AddSelection) {
         Gui::SelectionObject obj(sel);
         if (obj.getObject() == this->getObject()) {
+            if (this->getObject()->isDerivedFrom(Fem::FemPostPipeline::getClassTypeId()))
+                this->getObject()->Visibility.setValue(true);
+            this->getObject()->Visibility.setValue(true);
             if (this->getObject()->Visibility.getValue())
```

Comment:
If you set visibility to true, what is then the point on the next line to check whether the visibility is true or not?

Reply:
Oops, sorry. Seems a copy/paste mistake.
Now it is correct.